main.cpp

```diff
@@ -19,13 +19,25 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
+    CastCallBack(Rewriter& rewriter):_rewriter(rewriter) {
         // Your code goes here
     };
 
     void run(const MatchFinder::MatchResult &Result) override {
         // Your code goes here
+	auto cstmtptr = Result.Nodes.getNodeAs<clang::CStyleCastExpr>("cast");
+	auto l_brace = cstmtptr->getLParenLoc();
+	auto r_brace = cstmtptr->getRParenLoc();
+	auto end_of_var = Lexer::getLocForEndOfToken(cstmtptr->getSubExprAsWritten()->IgnoreImpCasts()->getEndLoc(), 0, *Result.SourceManager, LangOptions());

```

Comment:
Please describe why you are using here `getSubExprAsWritten()` and `IgnoreImpCasts()`.
What would be the difference with the use of `getSubExprAsWritten()->IgnoreImpCasts()`, `getSubExpr()->IgnoreImpCasts()` and `getSubExprAsWritten()`?

Reply:
Let`s change the code this way:
![image](https://user-images.githubusercontent.com/61185004/170837913-2696aef4-10dd-4860-93c3-df636eb395ff.png)

The output will be like this:

![image](https://user-images.githubusercontent.com/61185004/170837928-53d1a6ca-9553-4832-b01c-61892ee05649.png)

Judging by this output, it can be said that it is pointless to use getSubExprAsWritten()->IgnoreImpCasts() here. Instead of this we can use getSubExprAsWritten() or getSubExpr()->ignoreImpCasts(). Meanwhile, getSubExpr() won`t cause the result that we expect.

Here is what the documentation says about these functions, so that we can understand the difference between them and why they are used:
getSubExprAsWritten() retrieves the cast subexpression as it was written in the source;
getSubExpr() retrieves the first cast subexpression;
ignoreImpCasts() skips past any implicit casts which might surround this expression until reaching a fixed point. It skips the following AST nodes: ImplicitCastExpr, FullExpr.